### PR TITLE
Upgrade of azure-devops-node-api package

### LINKED ISSUE #310: vso-node-api was renamed to azure-devops-api
Dependency should be updated.

### PR DESCRIPTION
**Pull request by henrybeen**
Upgrade of azure-devops-node-api package

Fixes #310

### CODE REVIEW

**Comment by trevorsg** [comment]
Thanks for the PR. Having a look today.

**Comment by trevorsg** [comment]
I'm running into some issues with extension publish (getting a 500 error on the PUT request). Reaching out to the owners to see if it's a bug on their end.

**Comment by henrybeen** [comment]
What exactly is going wrong? Do you need any action from me at this point?

I have validated these few weeks changes ago against TFS2019 preview, with a privately build package of azure-devops-node-api master.

**Comment by trevorsg** [comment]
I'm not sure right now, but it seems like a bug in the Marketplace. Running `tfx extension publish` against an on-prem Azure DevOps Server results in a 500 error with the new API. I have engaged the marketplace team to investigate.

**Comment by henrybeen** [comment]
Okay, please let me know if you need anything from me please. Otherwise I'll wait

**Comment by kesane-msft** [comment]
@henrybeen , I am from the Marketplace team. I will be looking into this early next week. Hope that is ok.

File: app/exec/extension/_lib/publish.ts [hunk 1]
```diff
@@ -293,9 +295,7 @@ export class PackagePublisher extends GalleryBase {
 	 * @return Q.Promise that is resolved when publish is complete
 	 */
 	public publish(): Promise<GalleryInterfaces.PublishedExtension> {
-		const extPackage: GalleryInterfaces.ExtensionPackage = {
-			extensionManifest: fs.readFileSync(this.settings.vsixPath, "base64"),
-		};
+		const extPackage = fs.createReadStream(this.settings.vsixPath);
```
Comment (kesane-msft) [inline app/exec/extension/_lib/publish.ts]: OnPrem doesn't support using octet streams for the vsix package. You need to revert this (and related changes).

**Comment by kesane-msft** [comment]
@henrybeen we are seeing the issue of 500 because of using octet stream as payload. That's not supported on on-prem. Using the base64 payload should solve the issue.

**Comment by trevorsg** [comment]
@henrybeen I will take care of making the necessary changes.

**Comment by henrybeen** [comment]
That is (very) much appreciated! :D

**Comment by kaptainkommie** [comment]
Any update on when this fix will be merged to master?

**Comment by trevorsg** [comment]
Aiming for today.

**Comment by trevorsg** [comment]
Picking this up in a new PR

**Pull request closed without merging**